providers/clusters/provider.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusters
+
+import (
+	"github.com/go-logr/logr"
+
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"sigs.k8s.io/multicluster-runtime/pkg/clusters"
+	"sigs.k8s.io/multicluster-runtime/pkg/multicluster"
+)
+
+var _ multicluster.Provider = &Provider{}
+
+// Provider is a provider that only embeds clusters.Clusters.
+type Provider struct {
+	clusters.Clusters[cluster.Cluster]
+	log logr.Logger
+}
+
+// New creates a new provider that embeds clusters.Clusters.
+func New() *Provider {
+	p := new(Provider)
+	p.log = log.Log.WithName("clusters-cluster-provider")
+	p.Clusters = clusters.New[cluster.Cluster]()
+	p.Clusters.ErrorHandler = p.log.Error
+	return p
+}
```

Comment:
How would you use this `Provider` struct? Given that it creates a log with a specific name, you wouldn't embed it in a real provider, would you?

Reply:
One use is to show that `clusters.Clusters` is enough to get a valid provider so downstream can focus on their domain specific implementations by just embedding `clusters.Clusters`. 
Another use is testing - I have an operator using mcr and to inject different clusters to test against I had to use combine the `multi` and `single` providers. 
With a provider like this test cases are much simpler to write.